### PR TITLE
Simplify and Optimize Volumetric Fog

### PR DESCRIPTION
**Pull request by reduz**
* Always use temporal re-project, it just looks way better than with any other filter.
* By always using temporal re-project, the shadow-map reduction can be done away with, massively improving performance (3/4 times faster).
* Also by always using temporal re-project, the fog fade in lights can be reduced significantly, this practically eliminates fog leaks.
* Still, disadvantage of temporal re-project is update latency so..
* Made sure a Gaussian filter runs in XY after fog, this allows to keep stability and lower latency.

image

### CODE REVIEW

**Review by JFonS — APPROVED** [review]
Lots of removed code, very little new code... nice!

File: servers/rendering/renderer_rd/renderer_storage_rd.cpp [hunk 1]
```diff
@@ -5371,7 +5371,7 @@ RID RendererStorageRD::light_create(RS::LightType p_type) {
 	light.param[RS::LIGHT_PARAM_SHADOW_BIAS] = 0.02;
 	light.param[RS::LIGHT_PARAM_SHADOW_BLUR] = 0;
 	light.param[RS::LIGHT_PARAM_SHADOW_PANCAKE_SIZE] = 20.0;
-	light.param[RS::LIGHT_PARAM_SHADOW_VOLUMETRIC_FOG_FADE] = 1.0;
+	light.param[RS::LIGHT_PARAM_SHADOW_VOLUMETRIC_FOG_FADE] = 0.2;
```
Comment (clayjohn) [inline servers/rendering/renderer_rd/renderer_storage_rd.cpp:5374]: Default is 0.2 here, but 0.1 in light_3d.cpp

Comment (reduz) [reply servers/rendering/renderer_rd/renderer_storage_rd.cpp:5374]: good catch

**Review by clayjohn — APPROVED** [review]

**Comment by reduz** [comment]
> Lots of removed code, very little new code... nice!

Yeah the old method just looked much, much worse than the new method and it was a lot more expensive. With the new one you lose the real-timeness (there will always be a bit of fade because of the reprojection) but win massively in quality.

**Comment by reduz** [comment]
Should be good to merge when CI passes.

**Comment by clayjohn** [comment]
Thanks! :)

**Pull request merged**